### PR TITLE
Add entry for removal default value of request_context to changelog

### PR DESCRIPTION
**Pull request by hramezani**
https://github.com/urllib3/urllib3/pull/2232/files#r640945296

### CODE REVIEW

File: changelog/1897.removal.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed default value of ``None`` for the ``request_context`` parameter of ``urllib3.poolmanager.PoolManager.connection_from_pool_key``.
```
Comment (sethmlarson) [inline changelog/1897.removal.rst:1]: Since `PoolManager` is exposed at `urllib3.PoolManager` let's use that as our reference

Comment (hramezani) [reply changelog/1897.removal.rst]: Done!

**Comment by sethmlarson** [comment]
@hramezani Added a line that let's users gloss over this one since the "removals" section will likely get the most eyeballs on it. What do you think?

**Comment by hramezani** [comment]
> @hramezani Added a line that let's users gloss over this one since the "removals" section will likely get the most eyeballs on it. What do you think?

Great :+1:

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**